### PR TITLE
Complete CI flow 

### PR DESCRIPTION
**Pull request by rohithkrn**
Added complete CI flow which includes dockerfile setup, clang-format, build, binary tests, operator tests, inference test.

### CODE REVIEW

**Review by petrex — COMMENTED** [review]
What is the usage of this "rocm_agent_enumerator"?

File: docker/ubuntu-16.04-rocm171/rocm-base/target.lst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+gfx803
```
Comment (petrex) [inline docker/ubuntu-16.04-rocm171/rocm-base/target.lst]: Have we tested our repo on gfx803?

**Comment by rohithkrn** [comment]
"rocm_agent_enumerator" is used while building the docker image to detect
the type of platform.

File: docker/ubuntu-16.04-rocm171/rocm-base/target.lst (hunk 1 above)
Comment (rohithkrn) [reply docker/ubuntu-16.04-rocm171/rocm-base/target.lst]: No.

Comment (petrex) [reply docker/ubuntu-16.04-rocm171/rocm-base/target.lst]: let's keep gfx900 then?

Comment (rohithkrn) [reply docker/ubuntu-16.04-rocm171/rocm-base/target.lst]: Ok. Will change that.

File: docker/ubuntu-16.04-rocm171/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    rm -rf /var/lib/apt/lists/*
+
+ENV HCC_HOME=$ROCM_PATH/hcc
+ENV HIP_PATH=$ROCM_PATH/hip
+ENV OPENCL_ROOT=$ROCM_PATH/opencl
+ENV PATH="$HCC_HOME/bin:$HIP_PATH/bin:${PATH}"
+ENV PATH="$ROCM_PATH/bin:${PATH}"
+ENV PATH="$OPENCL_ROOT/bin:${PATH}"
+ADD ./docker/ubuntu-16.04-rocm171/target.lst /opt/rocm/bin
+#ADD ./docker/ubuntu-16.04-rocm171/rocm-base/rocm_agent_enumerator /opt/rocm/bin
```
Comment (petrex) [inline docker/ubuntu-16.04-rocm171/Dockerfile:66]: Do we need `rocm_agent_enumerator`?

Comment (rohithkrn) [reply docker/ubuntu-16.04-rocm171/Dockerfile:66]: Not anymore. It was originally there in "miopen-frameworks-docker" setup. But, I removed it and tested it. It worked fine. Now, "rocm-utils" takes care of its job.

**Review by petrex — APPROVED** [review]

**Pull request merged**